ComponentKit/Utilities/CKComponentActionInternal.h

```
@@ -21,50 +21,74 @@
 
 /** Base case, recursion should stop here. */
-void _CKTypedComponentActionTypeVectorBuild(std::vector<const char *> &typeVector, const _CKTypedComponentActionTypelist<> &list);
+void CKTypedComponentActionTypeVectorBuild(std::vector<const char *> &typeVector, const CKTypedComponentActionTypelist<> &list);
 
 /** 
  Recursion through variadic argument type unpacking. This allows us to build a vector of encoded const char * before
  any actual arguments have been provided. All of this is done at compile-time.
  */
 template<typename T, typename... Ts>
-void _CKTypedComponentActionTypeVectorBuild(std::vector<const char *> &typeVector, const _CKTypedComponentActionTypelist<T, Ts...> &list)
+void CKTypedComponentActionTypeVectorBuild(std::vector<const char *> &typeVector, const CKTypedComponentActionTypelist<T, Ts...> &list)
 {
   typeVector.push_back(@encode(T));
-  _CKTypedComponentActionTypeVectorBuild(typeVector, _CKTypedComponentActionTypelist<Ts...>{});
+  CKTypedComponentActionTypeVectorBuild(typeVector, CKTypedComponentActionTypelist<Ts...>{});
 }
 
 /** Base case, recursion should stop here. */
-void _CKConfigureInvocationWithArguments(NSInvocation *invocation, NSInteger index);
+void CKConfigureInvocationWithArguments(NSInvocation *invocation, NSInteger index);
 
 /**
  Recursion here is through normal variadic argument list unpacking. Unlike above, we have the arguments, so we don't
  require the intermediary struct.
  */
 template <typename T, typename... Ts>
-void _CKConfigureInvocationWithArguments(NSInvocation *invocation, NSInteger index, T t, Ts... args)
+void CKConfigureInvocationWithArguments(NSInvocation *invocation, NSInteger index, T t, Ts... args)
 {
   // We have to be able to handle methods that take less than the provided number of arguments, since that will cause
   // an exception to be thrown.
   if (index < invocation.methodSignature.numberOfArguments) {
     [invocation setArgument:&t atIndex:index];
```

Yeah, so this works fine for plain data structs, primitives, and pointers, but more complex data types just silently fail here. If I get an IMP and cast it to the required type then I'm pretty sure it would work with complex types, but then I have other problems to deal with such as zeroing parameters that aren't provided in the send call (which we actually do, it's terrible). Suggestions? I considered just restricting types to NSObjects which wouldn't be that hard, but would still force us to use all these value wrappers everywhere we have right now.